include/oneapi/dpl/pstl/parallel_backend_utils.h

```diff
@@ -186,7 +188,7 @@ __set_intersection_construct(_ForwardIterator1 __first1, _ForwardIterator1 __las
         {
             if (!__comp(*__first2, *__first1))
             {
-                ::new (::std::addressof(*__result)) _Tp(*__first1);
+                ::new (::std::addressof(*__result)) _Tp(*__select_copy(__first1, __first2));
```

Comment:
As for me we able to make this line better (to increase performance).
For example I mean we may take a look to another place in our code:
```
        auto __move_values = [](_RandomAccessIterator __x, _Tp* __z) {
            if constexpr (::std::is_trivial_v<_Tp>)
                *__z = ::std::move(*__x);
            else
                ::new (::std::addressof(*__z)) _Tp(::std::move(*__x));
        };
```

Reply:
I don't think we have any guarantees that the selected iterator's `value_type` matches the `value_type` of `__result`.  I think we only know require that it can be converted to the `value_type` of result. 

Also, I don't think we should use `std::move` on input data.  This would move instead of copy from the input set to the output set, which changes the semantics of `std::set_intersection`.

If we adapt this code by removing `::std::move()` in both locations, I think it could fit.  However, would that improve performance to assign for trivial types rather than copy construct?

Reply:
Sure,  I can add it, but without the `std::move`.

Reply:
I've added this.  Is this what you had in mind?

Reply:
Yes. thank you!